Repomanage tests, add srpm to modules, remove redundant packages (RhBug:1804720)

kontura opened the pull request:
While trying to have saner repomanage output I have noticed there are several unused packages which only make things more complicated.

There are also various fixes for modular metadata and modular packages.

https://bugzilla.redhat.com/show_bug.cgi?id=1804720

pkratoch commented on dnf-behave-tests/fixtures/specs/dnf-ci-fedora-modular/postgresql-9.6.8-1.module_1710+b535a823.spec:
```diff
@@ -26,20 +26,6 @@ PostgreSQL server, or on a remote machine that accesses a PostgreSQL server
-The postgresql-contrib package contains various extension modules that are
-included in the PostgreSQL distribution.
-
 %package devel
```
This subpackage should be also removed to stay consistent.
Also, all "Requires: postgresql-devel" should be removed from the postgresql-test package.

pkratoch commented on dnf-behave-tests/features/plugins-core/repomanage.feature:
```diff
@@ -0,0 +1,202 @@
+
+
+@bz1804720
+Scenario: all packages get listed as new if --keep is really big
```
This is just a nitpicking, but can you please substitute "really" for "sufficiently"? Because "really" is really vague and kind of sounds like there is some magic number above which all packages get listed.

kontura commented on dnf-behave-tests/features/plugins-core/repomanage.feature:
```diff
@@ -0,0 +1,202 @@
+
+Scenario: basic functionality of repomanage --new works without repodata
+Given I copy repository "dnf-ci-thirdparty-updates" for modification
+  And I delete directory "/{context.dnf.repos[dnf-ci-thirdparty-updates].path}/repodata"
```
It actually has to be there, this way the path start with `//` which means it [doesn't get prepended with installroot](https://github.com/rpm-software-management/ci-dnf-stack/blob/master/dnf-behave-tests/common/lib/file.py#L24), that is what we need since the `{context.dnf.repos[dnf-ci-thirdparty-updates].path}` already contains the installroot path.

lukash replied:
I really dislike this btw. Thinking about it maybe we could prepend installroot only if the path doesn't start with a format placeholder. Because I think it almost always should if it's not an installroot path.

kontura replied:
I agree, that could work, paths not starting with format placeholder really shouldn't be pointing to the host and thus can be prepended.

But this pr is complex enough, so I don't want to do this kind of change here.

pkratoch replied:
You are right, my bad.

kontura commented:
> Also, this is not related to the bug, but since you are also adding the basic tests for repomanage, could you also add a scenarios for the case when there are multiple files for the newest nevra and test that `repomanage --new` will output both files and `--repomanage new -k 2` will output both files and the next one? Because this makes repomanage different from just returning the latest packages.

I have added two new tests with duplicate packages, it turned out there was actually a bug in there, good catch. :+1: 

> Can you please also add a scenario for case when there are both modular and non-modular packages of the same name in the repo?
> 
Changed repo to `dnf-ci-fedora-modular-updates` for `Scenario: list packages as new if they are in different stream or are non modular`, because it contains a non-modular nodejs as well as several other streams of nodejs.

pkratoch commented:
> I have added two new tests with duplicate packages, it turned out there was actually a bug in there, good catch. +1

Well, now I just hope this indeed was the intended behavior.

Thanks for adding the tests, looks good to me.
